### PR TITLE
chore: Semantic release

### PR DESCRIPTION
**Pull request by sunnieeeYoon**
### Description of change
- applying semantic-release
- adding 'breaking' and 'ci' to Commit Convention.
- applying github action
스크린샷 2024-01-31 오후 4 33 57

### Pull-Request Checklist

- [x] Code is up-to-date with the `main` branch
- [ ] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/JunbaeJs/junbae-mode/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,5 +59,147 @@
         "title": "Hello World"
       }
     ]
+  },
+  "release": {
+    "branches": [
+      {
+        "name": "master"
```
Comment (changchanghwang) [inline package.json]:
```suggestion
        "name": "main"
```

File: package.json [hunk 2]
```diff
@@ -58,5 +59,147 @@
...
+      },
+      {
+        "name": "dev",
+        "prerelease": "dev"
```
Comment (changchanghwang) [inline package.json]:
```suggestion
        "name": "develop",
        "prerelease": "develop"
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -3,11 +3,13 @@
...
 ```
+breaking: top-level version bump. used for major framework changes or large-scale project deployments.
 feat: new feature for the user, not a new feature for build script
 fix: bug fix for the user, not a fix to a build script
 docs: changes to the documentation
 style: formatting, missing semi colons, etc; no production code change
 refactor: refactoring production code, eg. renaming a variable
 test: adding missing tests, refactoring tests; no production code change
 chore: updating grunt tasks etc; no production code change
+ci: ci update
```
Comment (changchanghwang) [inline CONTRIBUTING.md]: We don't need this. Please use `chore:`

File: package.json [hunk 4]
```diff
@@ -58,5 +59,147 @@
...
+              {
+                "type": "chore",
+                "section": "📦 Chores",
+                "hidden": true
+              },
+              {
+                "type": "ci",
+                "section": "💫 CI/CD",
+                "hidden": false
+              },
```
Comment (changchanghwang) [inline package.json]: We don't need this - 2.

File: package.json [hunk 5]
```diff
@@ -58,5 +59,147 @@
...
+              {
+                "type": "test",
+                "section": "✅ Tests",
+                "hidden": true
+              }
```
Comment (changchanghwang) [inline package.json]: add breaking plz

File: package.json [hunk 6]
```diff
@@ -58,5 +59,147 @@
         "title": "Hello World"
       }
     ]
+  },
+  "release": {
```
Comment (changchanghwang) [inline package.json]: Please move this all changes to `.releaserc`

**Review by myunggyuns — APPROVED** [review]

File: .github/workflows/cicd.yml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+permissions:
+  contents: read # for checkout
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [20.11.x]
```
Comment (changchanghwang) [inline .github/workflows/cicd.yml]: How about 20.x ?

File: .github/workflows/cicd.yml
Comment (changchanghwang) [inline .github/workflows/cicd.yml:1]: change File name plz. It Not include `cd`

File: .github/workflows/cicd.yml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            ${{ runner.OS }}-
+      - name: Package install
+        run: npm ci
+      - name: lint 
+        run: |
+          npm run lint
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (changchanghwang) [inline .github/workflows/cicd.yml]: I think it's not necessary for us. It means token for NPM registry

File: .releaserc [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+    "plugins": [
+      [
+        "@semantic-release/commit-analyzer",
+        {
+          "preset": "angular",
+          "releaseRules": [
+            {
+              "scope": "no-release",
+              "release": false
+            },
```
Comment (changchanghwang) [inline .releaserc]: What is this?

File: .releaserc [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+              "release": "patch"
+            },
+            {
+              "type": "chore",
+              "release": false
+            },
+            {
+              "type": "ci",
+              "release": false
+            },
```
Comment (changchanghwang) [inline .releaserc]: remove plz

File: CHANGELOG.md
Comment (changchanghwang) [inline CHANGELOG.md:1]: ?

File: package.json [hunk 11]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "junbae-mode",
   "displayName": "Junbae Mode",
-  "version": "0.0.1",
+  "version": "1.0.2",
```
Comment (changchanghwang) [inline package.json]: ?

File: package.json [hunk 12]
```diff
@@ -58,5 +59,11 @@
         "title": "Hello World"
       }
     ]
+  },
+  "dependencies": {
+    "@semantic-release/changelog": "^6.0.3",
+    "@semantic-release/commit-analyzer": "^10.0.1",
+    "@semantic-release/git": "^10.0.1",
+    "@semantic-release/release-notes-generator": "^11.0.4"
   }
```
Comment (changchanghwang) [inline package.json]: It's not dependencies. move to devDep

File: .github/workflows/cicd.yml (hunk 7 above)
Comment (sunnieeeYoon) [reply .github/workflows/cicd.yml]: 스크린샷 2024-02-01 오후 3 51 50
we require Node version 20.8.1 or higher. and LTS is 20.11.0

**Review by changchanghwang — APPROVED** [review]
LGTM

**Pull request merged**